src/unit_tests/os_crypto/sha256/test_sha256_op.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2015-2021, Wazuh Inc.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <cmocka.h>
+
+#include "../../os_crypto/sha256/sha256_op.h"
+#include "../../wrappers/common.h"
+#include "../headers/shared.h"
+
+// Tests
+
+void test_sha256_string() {
+    const char *string = "teststring";
+    const char *string_sha256 = "3c8727e019a42b444667a587b6001251becadabbb36bfed8087a92c18882d111";
+    os_sha256 buffer;
+
+    OS_SHA256_String(string, buffer);
+
+    assert_string_equal(buffer, string_sha256);
+}
+
+void test_sha256_string_sized() {
+    const char *string = "teststring";
+    const char *string_sha256_chopped = "3c8727e019";
+    int chopped_size = 10;
+    os_sha256 buffer;
+
+    OS_SHA256_String_sized(string, buffer, chopped_size);
+
+    assert_string_equal(buffer, string_sha256_chopped);
+}
+
+void test_sha256_file() {
```

Comment:
Here, we need to wrap all the necessary methods to make  `OS_SHA256_File()` thinks it's accessing a real file.
We shouldn't create a temporary file.